app/scripts/directives/oscPersistentVolumeClaim.js

```diff
@@ -0,0 +1,71 @@
+"use strict";
+
+angular.module("openshiftConsole")
+  .directive("oscPersistentVolumeClaim", function($filter){
+    return {
+      restrict: 'E',
+      scope: {
+        claim: "=model",
+        showNameInput: "=",
+        showAmountInput:"="
+      },
+      templateUrl: 'views/directives/osc-persistent-volume-claim.html',
+      link: function(scope) {
+              var amountAndUnit = $filter('amountAndUnit');
+              var humanizeUnit = $filter('humanizeUnit');
+
+              // Create a unique ID for `label for` and `aria-describedby` attributes.
+              scope.id = _.uniqueId('storage-size-');
+
+              // If unit is not already in options, add it.
+              var addUnitOption = function(unit) {
+                if (!_.some(scope.units, { value: unit })) {
+                  scope.units.push({
+                    value: unit,
+                    label: humanizeUnit(unit, scope.type)
+                  });
+                }
+              };
+
+              scope.$watch('defaultValue', function(defaultValue) {
```

Comment:
There's no default value for PVCs unlike resource limits. You should be able to remove this code.